### PR TITLE
Add cluster size continuous helper method

### PR DESCRIPTION
**Pull request by elenatorro**
Related issue #654 

Add `cluster_size_legend` method

Examples http://localhost:8888/notebooks/cartoframes/examples/04_helper_methods/03_helper_methods_part_3.ipynb#

Screenshot 2019-08-13 at 10 16 31
Screenshot 2019-08-13 at 10 16 24
Screenshot 2019-08-13 at 10 26 45

### CODE REVIEW

**Comment by makella** [comment]
hiii @elenatorro ok I spent some time today going through the new method (thanks again for getting it going!).

I would like to spend some more time thinking through a few things, but with this commit:
- [x] simple change to the helper method name removing continuous it
- [x] I changed the ramp so the legend draws where the sqrt is on the symbol size. I want to discuss a couple of points here when we meet (sqrt on the symbol sizes vs. sqrt on all parts of the equation like in the size continuous layer, but then this impacts returned legend values which is something that I need to look back at for the size continuous layer again but at the same time, I feel that is the right way to do it but that will have an impact on legend values)
- [x] I added some additional math to the first break point size (good idea!) because the smallest symbol size wasn't working well across different resolutions

Other notes to look at/discuss:
- [ ] if widget=True right now would default to a histogram widget. I'm getting an error because I think `value` is being read in as `$clusterOperation` with the dollar sign because it is looking for an attribute value vs. clusterCount(). 
  - To think about/discuss also here is if we should have a formula widget or histogram widget, or no widget 
- [ ] when animating on another attribute, the time-series widget isn't loading. this makes sense because we are clustering by count, but think it will be good if we can talk through this as well
- since we have `clusterOperation` does this mean that we can have the cluster type be a parameter also (for example clusterAvg, Min, Max, etc.)

**Comment by elenatorro** [comment]
Helper updated, let's go through your questions / suggestions:

> since we have clusterOperation does this mean that we can have the cluster type be a parameter also (for example clusterAvg, Min, Max, etc.)

Yes, with the last changes we can. By default, the operation would be `count`, but you can specify a different operation:

```py
from cartoframes.viz.helpers import cluster_size_layer

Map(
    cluster_size_layer('vancouver_trees', 'avg', 'diameter')
)
```

> when animating on another attribute, the time-series widget isn't loading. this makes sense because we are clustering by count, but think it will be good if we can talk through this as well

I don't know if you should be add animation in this helper, let's discuss it later 👍

> widget

Now, the histogram widget works for all the cluster operations except for the `count` one, since it doesn't depend on a variable. I'm gonna try later to fix it (although it's not trivial), but maybe adding a formula widget would be better for this case

* Others

- Popup + legend values changed, we need to discuss which format we want to use to display it (like: diameter (avg), (Avg) diameter...)

File: cartoframes/viz/helpers/cluster_size_layer.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        source, operation='count', value=None, resolution=32,
+        title='', size=None, color=None, description='', footer='',
+        legend=True, popup=True, widget=False, animate=None):
+    """Helper function for quickly creating a size symbol map with
+    continuous size scaled by cluster.
+
+    Args:
+        source (:py:class:`Dataset <cartoframes.data.Dataset>` or str): Dataset
+          or text representing a table or query associated with user account.
+        value (str): Column to symbolize by.
```
Comment (oleurud) [inline cartoframes/viz/helpers/cluster_size_layer.py]: `operation` argument documentation missing

File: cartoframes/viz/helpers/cluster_size_layer.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+def _get_cluster_operation_title(operation, value):
+    if value is not None and operation != 'count':
+        return '{0} ({1})'.format(value, operation)
+
+    return operation
+
+
+def _get_cluster_operation(operation, value):
+    if value is not None and operation != 'count':
```
Comment (oleurud) [inline cartoframes/viz/helpers/cluster_size_layer.py]: Should we also check if `operation` exists in `CLUSTER_OPERATIONS` ?

**Pull request merged**